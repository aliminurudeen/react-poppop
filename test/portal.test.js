import React from 'react';
import Portal from '../src/portal';
import {shallow, render, mount} from 'enzyme';
import toJson from 'enzyme-to-json';

it('remove overflow style after unmount', () => {
  expect(document.body.style.overflow).toEqual('');
  const component = mount(<Portal><div>hi</div></Portal>);
  expect(document.body.style.overflow).toEqual('hidden');
  component.unmount();
  expect(document.body.style.overflow).toEqual('');
  expect(component.html()).toEqual(null);
})

it('render', () => {
  const component = mount(<Portal><div>hi</div></Portal>);
  expect(toJson(component)).toMatchSnapshot();
  jest.clearAllMocks();
});
