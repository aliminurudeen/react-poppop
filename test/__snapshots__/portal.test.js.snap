// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<Portal>
  <div>
    hi
  </div>
</Portal>
`;
